lib/pbench/server/api/resources/query_apis/metadata_index/timeseries_samples.py

```diff
@@ -0,0 +1,232 @@
+                            the result.
+
+                "filters": key-value representation of query filter parameters to
+                           narrow the search results e.g. {"sample.name": "sample1"}
+
+        EXAMPLES:
+            {
+                "run_id": "1234567"
+                "filters": {"sample.name": "sample1"},
+            }
+            or
+            {
+                "run_id": "1234567"
+                "scroll_id": "cmFuZG9tX3Njcm9sbF9pZF9zdHJpbmdfMg=="
+            }
+        """
+        run_id = context["run_id"]
+        dataset = context["dataset"]
+        scroll_id = json_data.get("scroll_id")
+
+        self.logger.info(
+            "Return TimeSeries sample rows {} for dataset {}, prefix {}, "
+            "run id {} next page "
+            if scroll_id
+            else "",
+            dataset,
+            self.prefix,
+            run_id,
+        )
+
+        scroll_id = json_data.get("scroll_id")
+        if scroll_id:
+            return {
+                "path": "/_search/scroll",
+                "kwargs": {
+                    "json": {
+                        "scroll": TimeSeriesSamples.SCROLL_EXPIRY,
+                        "scroll_id": scroll_id,
+                    }
+                },
+            }
+
+        # Retrieve the ES indices that belong to this run_id
+        indices = self.get_index(dataset, "result-data")
+        if not indices:
+            self.logger.debug(
+                f"Found no indices matching the prefix result-data"
+                f"for a dataset {dataset!r}"
+            )
+            abort(HTTPStatus.NOT_FOUND, message="Found no matching indices")
+
+        try:
+            # Retrive result-data mappings for validating user provided filters
+            mappings = self.get_mappings("result-data")
+        except (TemplateNotFound, KeyError) as e:
+            self.logger.exception(f"Exception while getting 'result-data' template {e}")
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+
+        es_filter = [{"match": {"run.id": run_id}}]
+        query_strings = []
+
+        # Validate each user provided filters against the result-data document
+        # mappings. If the filter is not present in result-data mappings
+        # properties we will not apply that filter in the ES query.
+        for filter, value in json_data.get("filters", {}).items():
+            if filter in mappings.keys() and mappings[filter] == "keyword":
+                # Get all the keyword filters to apply
+                es_filter.append({"match": {filter: value}})
+            elif filter in mappings.keys() and mappings[filter] != "keyword":
+                # Get all the text filters to apply
+                # Note: There is only one text field sample.measurement_title
+                # in result-data documents and if we can re-index it as a
+                # keyword we can get rid of this loop.
```

Comment:
Hmm. Of course in the `result-data-sample` mappings, I'd overlaid `measurement_title.raw` as a keyword; I found that out from working with real Dashboard queries, but getting to the `result-data` index data is a bit more complex and I guess I didn't realize at the time that the field was duplicated.

We could set up the same mapping there, of course, because we're still going to be re-indexing from scratch on our deployments for a while. However, presuming a "namespace" query, it probably wouldn't even be returning a non-keyword field as filterable, so you shouldn't be able to get here. 🤔

Reply:
But `"filters"` defines which timeseries data points to return, not which items from each datapoint are revealed in the `"_source"` document. Yes, we could *also* allow specifying which non-keyword fields to return to the caller rather than returning all of them, but that's got nothing to do with `"filters"`.

Reply:
Oh, dear...of course you are right:  there are _two_ issues here -- what _documents_ do we want to select, and what _fields_ do we **fetch** from those documents.  And, they both face similar questions.

In the case of document selection, we're (probably) principally interested in `keyword`/text fields (although, if the data fields aren't in the document, it might be handy to avoid selecting it!), and if the request references fields which are not in the mapping, we should probably report an error; however, in the case of selecting fields to return, we're probably interested in the numerical fields rather than the text ones, and it's probably OK if they aren't present in the selected documents.

Nevertheless, it probably would be valuable to be able to stipulate which data fields get returned (i.e., to avoid returning lots of stuff that the client doesn't want).  I dunno...are the different series indexed in separate documents?  (I.e., if I ask for latency data, I'm not going to inadvertently get throughput fields as well?)

Reply:
Yeah if we can update the `measurement_title` mapping to keyword with our new deployment that would be great.